bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,201 @@
+            parsed_info["brief_comment"] = cursor.brief_comment
         if cursor.raw_comment:
-            holder["raw_comment"] = cursor.raw_comment
-        holder["members"] = []
-
-        parsed_list.append(holder)
+            parsed_info["raw_comment"] = cursor.raw_comment
+        parsed_info["members"] = []
 
+    # Get cursor's children and recursively add their info to the dictionary, as members of the parent
     for child in cursor.get_children():
-        if node_in_this_file(child, this_filename):
-            child_list = []
-            generate_parsed_info(
-                child, filter, lines, this_filename, depth + 1, child_list,
-            )
-            if child_list and parsed_list:
-                if len(parsed_list[0]["members"]):
-                    parsed_list[0]["members"].append(child_list[0])
-                else:
-                    parsed_list[0]["members"] = child_list
+        child_node = {"cursor": child, "filename": filename, "depth": depth + 1}
+        # Check if the child belongs to the file
+        if is_node_in_this_file(**child_node):
+            child_parsed_info = generate_parsed_info(**child_node)
+            # If both child and parent's info is not empty, add child's info to parent's members
+            if child_parsed_info and parsed_info:
+                parsed_info["members"].append(child_parsed_info)
 
+    return parsed_info
 
-def parse_arguments(args):
-    parser = argparse.ArgumentParser(description="C++ libclang parser")
-    parser.add_argument("files", nargs="+", help="The source files to search")
-    return parser.parse_args(args)
 
+def get_compilation_commands(compilation_database_path, filename):
+    """
+    Returns the compilation commands extracted from the the compile command
 
-def get_output_path(source, output_dir):
-    x_list = source.split("pcl/", 1)[-1]
-    x_list = x_list.split("/")
-    extra = ["pcl", "include"]
+    Arguments:
+        - compilation_database_path: The path to `compile_commands.json`
+        - filename: The file's name to get it's compilation commands
 
-    filename = x_list[-1].split(".")[0]
-    relative_dir = "/".join(x for x in x_list[:-1] if x not in extra)
-    dir = os.path.join(output_dir, relative_dir)
+    Returns:
+        - compilation commands (list): The arguments passed to the compiler
+    """
 
-    # ensure the new directory exists
-    if not os.path.exists(dir):
-        os.makedirs(dir)
+    # Build a compilation database found in the given directory
+    compilation_database = clang.CompilationDatabase.fromDirectory(
+        buildDir=compilation_database_path
+    )
 
-    return f"{dir}/{filename}.json"
+    # Get compiler arguments from the compilation database for the given file
+    compilation_commands = compilation_database.getCompileCommands(filename=filename)
 
+    # Extracting argument list from the command's generator object
+    compilation_commands = list(compilation_commands[0].arguments)[1:-1]
```

Comment:
Let me ask about the reasoning behind 0, 1 and -1. You've added comments of **What** you're doing (eg: `Get compiler arguments from the compilation database for the given file`), but you're missing **Why** you're doing that thing in this way.

Eg:
* Why you didn't use `compile_commands, _ = db.getCompileCommands(name)`
* Why are the first and last commands useless?

Reply:
Added:
```
    """
    - compilation_commands:
        - An iterable object providing all the compilation commands available to build filename.
        - type: <class 'clang.cindex.CompileCommands'>
    - compilation_commands[0]:
        - Since we have only one command per filename in the compile_commands.json, extract 0th element
        - type: <class 'clang.cindex.CompileCommand'>
    - compilation_commands[0].arguments:
        - Get compiler arguments from the CompileCommand object
        - type: <class 'generator'>
    - list(compilation_commands[0].arguments)[1:-1]:
        - Convert the generator object to list, and extract compiler arguments
        - 0th element is the compiler name
        - nth element is the filename
    """
```